### PR TITLE
Update the README

### LINKED ISSUE #121: Cleanup and update the README
**Is your feature request related to a problem? Please describe.**
Right now README contains information that are outdatated and that are not related to the actual Model Registry target

**Describe the solution you'd like**
* Cleanup outdatated information. e.g., regarding database migration and server startup
* Provide additional information on:
  * Brief description of what Model Registry is and how the MLMD interaction is implemented
  * How to startup the rest proxy
  * How to build the project
  * How to run the tests
  * How to build the Docker image

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by lampajr**
Fixes https://github.com/opendatahub-io/model-registry/issues/121

## Description

Updated the README based on the actual target/focus of the project.

I also added two `docker compose` files that could make the startup of ml-metadata and model-registry easier.

## How Has This Been Tested?

Tested running: 

```shell
docker compose -f docker-compose.yaml up
```

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by lampajr** [comment]
As highlighted in the PR description I added two docker compose files which can make the local startup easier, if you think they are not needed at all I can simply remove them.

File: README.md [hunk 1]
```diff
@@ -2,96 +2,106 @@
...
-```
-The generated binary uses spf13 cmdline args. More information on using the server can be obtained by running the command:
-```
-./model-registry --help
-```
+
 ## OpenAPI Proxy Server
+
+The model registry proxy server implementation follows a contract-first approach, where the contract is identified by [model-registry.yaml](api/openapi/model-registry.yaml) OpenAPI specification.
+
```
Comment (tarilabs) [inline README.md:17]:
```suggestion
The model registry proxy server implementation follows a contract-first approach, where the contract is identified by [model-registry.yaml](api/openapi/model-registry.yaml) OpenAPI specification.

You can also easily display the latest OpenAPI contract for model-registry in a Swagger-like editor directly from this repository; for example, [here](https://editor.swagger.io/?url=https://raw.githubusercontent.com/opendatahub-io/model-registry/main/api/openapi/model-registry.yaml).
```

File: README.md [hunk 2]
```diff
@@ -2,96 +2,106 @@
...
+> **NOTE:** Docker compose must be installed in your environment.
+
+There are two `docker-compose` files that make the startup of both model registry and ml-metadara easier, by simply running:
+
+```shell
+docker compose -f docker-compose[-local].yaml up
+```
 
-Once the server has started, test clients and playground can be used as described in the above sections. 
+The main difference between the two docker compose files is that `-local` one build the model registry from source, the other one, instead, download the `latest` pushed [quay.io](https://quay.io/repository/opendatahub/model-registry?tab=tags) image.
```
Comment (tarilabs) [inline README.md]:
```suggestion
The main difference between the two docker compose files is that `-local` one build the model registry from source, the other one, instead, download the `latest` pushed [quay.io](https://quay.io/repository/opendatahub/model-registry?tab=tags) image.

When shutting down the docker compose, you might want to clean-up the SQLite db file generated by ML Metadata, for example `./test/config/ml-metadata/metadata.sqlite.db`
```

**Review by tarilabs — DISMISSED** [review]
Awesome, thank you.

Couple of minor suggestions for your consideration

**Review by tarilabs — APPROVED** [review]

**Pull request merged**